core/src/Kokkos_View.hpp

```diff
@@ -786,8 +879,11 @@ class View : public ViewTraits<DataType, Properties...> {
   }
   KOKKOS_INLINE_FUNCTION
   const Kokkos::Impl::SharedAllocationTracker& impl_track() const {
-    return m_track;
+    return m_track.m_tracker;
   }
+
+  KOKKOS_INLINE_FUNCTION
+  const view_tracker_type& impl_view_track() const { return m_track; }
```

Comment:
If you're adding these, why do you need the friend declaration? Also, it's completely unclear from the names what the differences are between `impl_track()` and `impl_view_track()`.  Ideally, more descriptive names here would be good, but if not, at least some documentation.

Reply:
impl_track is needed for backward compatibility.  impl_view_track was used in ViewTracker constructor/assign.  I fixed the friend clause and removed this function in favor or using m_track directly